url026791d51b_dhleong_SeriesTracker_v6_tgz-pJ8-stv6_SeriesTrackerJ8/src/stv6/episodes/managers/MediaTombManager.java

```diff
@@ -0,0 +1,232 @@
+			NodeList nodes = cont.getElementsByTagName("container");
+			Node curr;
+	        for (int i=0; i<nodes.getLength(); i++) {
+	        	curr = nodes.item(i);
+
+	    		// only extend the series name if this one has episodes in it
+	    		String newName = path.localDirectory + File.separator + curr.getTextContent();        			
+	    		String newId = curr.getAttributes().getNamedItem("id").getNodeValue();
+	    		subpaths.add( new Path(newId, newName) );
+	        }
+		} catch (SAXException e) {
+		} catch (IOException e) {
+		}
+		
+		return subpaths;
+	}
+	
+	/**
+	 * @return True if the MediaTomb is updating its database 
+	 */
+	public boolean isUpdating() {
+		try {
+			Document doc = loadPage("update&force_update=0");
+			NodeList tasklist = doc.getElementsByTagName("task");
+			return (tasklist.getLength() > 0);			
+		} catch (SAXException e) {
+			e.printStackTrace();
+		} catch (IOException e) {
+		}
+		return false;
+	}
+	
+	/**
+	 * @param url, starting with the req_type=
+	 * 	eg: "containers&parent_id=0"
+	 * @return
+	 * @throws SAXException
+	 * @throws IOException
+	 */
+	private Document loadPage(String url) throws SAXException, IOException {		
+		DOMParser parser = new DOMParser();
+		parser.parse("http://"+broadIp+":"+mtPort+"/content/interface?sid="+sid+"&req_type="+url);
+		Document doc = parser.getDocument();
+		return doc;
+	}
+	
+	/**
+	 * @return True on success, else false
+	 */
+	private boolean refreshSID() {
+		// get an sid for mediatomb
+		try {
+			Document siddoc = loadPage("auth&checkSID=1");
+			NodeList rootlist = siddoc.getElementsByTagName("root");
+			Node rootNode = rootlist.item(0);			
+			sid = rootNode.getAttributes().getNamedItem("sid").getNodeValue();			
+		} catch (SAXException e) {
+			e.printStackTrace();
+			return false;
+		} catch (IOException e) {
+			//e.printStackTrace();
+			System.out.println("Could not connect to mediatomb server. Failed.");
+//			System.exit(1);		
+			Profile.getInstance().unselect();
+			return false;
+		} catch (NullPointerException e) {
+			System.out.println(" * Could not retrieve a new SID.");
+			return false;
+		}
+		
+		return true;
+	}
+
+	@Override
+	public void reload() {
+		refreshSID();
+		
+		// wait until it's done updating
+		if (isUpdating()) {
+			System.out.print("EM: Waiting for MediaTomb to finish updating");
+			STHandlerManager.getInstance().setStaticMessage(true);
+			StaticMessageHandler.getInstance().setBody("Waiting for MediaTomb to finish updating...");
+			do {
+				doSleep(UPDATE_WAIT);
+				System.out.print(".");
+			} while (isUpdating());
+			System.out.println(" Done.");
+			StaticMessageHandler.getInstance().setBody();
+			STHandlerManager.getInstance().setStaticMessage(false);
+		}
+	}
+	
+	private class SubtitleMatchingFilter implements FilenameFilter {
+		
+		private final String original;
+		
+		public SubtitleMatchingFilter(String original) {
+			this.original = original.substring(0, original.lastIndexOf('.'));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If the lastIndexOf() search on line 222 does not find the substring and returns -1, the call to substring() on line 222 will throw IndexOutOfBoundsException. Note that the method is publicly accessible to external callers who might not be aware of the constraint on string original.